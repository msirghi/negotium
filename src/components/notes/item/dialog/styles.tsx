import { makeStyles } from '@mui/styles';

export const useNoteDialogStyles = makeStyles({
  container: {
    height: '50vh',
  },
  descriptionContainer: {},
  closeIcon: {
    position: 'absolute',
    right: 25,
    top: 25
  },
  titleContainer: {
    position: 'relative',
  },
});
