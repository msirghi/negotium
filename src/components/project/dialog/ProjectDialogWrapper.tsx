import { FC } from 'react';
import { ProjectDialog } from '../../common/siteWrapper/lists/wrapper/projectDialog/ProjectDialog';
import { Project } from '../../../common/types/projects.types';
import ProjectService from '../../../services/ProjectService';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from '../../../redux/store';
import { setProjectsList } from '../../../redux/projects/projectsSlice';

type Props = {
  project: Project;
  open: boolean;
  setOpen: (val: boolean) => void;
};

export const ProjectDialogWrapper: FC<Props> = ({ project, open, setOpen }) => {
  const dispatch = useDispatch();
  const projects = useSelector((state: RootState) => state.projects.projects);

  const onSubmit = async (title: string, color: Project['color']) => {
    await ProjectService.updateProjectName(project.id, title);
    await ProjectService.updateProjectColor(project.id, color);
    const updatedProject = { ...project, name: title };
    const updatedProjects = projects.map((p) =>
      p.id === project.id ? updatedProject : p
    );
    dispatch(setProjectsList(updatedProjects));
  };

  return (
    <ProjectDialog
      selectedProject={project}
      dialogTitle={'Edit Project'}
      onSubmit={onSubmit}
      open={open}
      setOpen={setOpen}
      submitButtonTitle={'Edit'}
    />
  );
};
