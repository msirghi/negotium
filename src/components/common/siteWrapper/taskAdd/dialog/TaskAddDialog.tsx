import Dialog from '@mui/material/Dialog';
import DialogContent from '@mui/material/DialogContent';
import { TaskAddButton } from '../../../content/taskWrapper/section/taskAdd/TaskAddButton';
import { Nullable } from '../../../../../common/types/common.types';
import { useTasksActions } from '../../../../../common/hooks/tasks/useTasksActions';
import { makeStyles } from '@mui/styles';

type Props = {
  open: boolean;
  handleClose: () => void;
};

const useStyles = makeStyles({
  dialog: {
    position: 'absolute',
    top: 50,
  },
});

export const TaskAddDialog = ({ open, handleClose }: Props) => {
  const { handleTaskAdd } = useTasksActions();
  const classes = useStyles();

  const onTaskAdd = (title: string, date: Nullable<Date>) => {
    handleTaskAdd(title, date);
    handleClose();
  };

  return (
    <Dialog
      open={open}
      onClose={handleClose}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
      fullWidth
      hideBackdrop
      classes={{ paper: classes.dialog }}
    >
      <DialogContent>
        <TaskAddButton defaultOpened onTaskAdd={onTaskAdd} cancelCallback={handleClose} />
      </DialogContent>
    </Dialog>
  );
};
