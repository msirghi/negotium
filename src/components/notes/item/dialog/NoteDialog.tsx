import { Note } from '../../../../common/types/notes.types';
import { FC, useState } from 'react';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  IconButton,
} from '@mui/material';
import { BorderlessInput } from '../../../common/form/input/borderlessInput/BorderlessInput';
import { Box } from '@mui/system';
import { useNoteDialogStyles } from './styles';
import RichTextField from '../../../common/form/input/richText/RichTextField';
import { InputChangeEvent } from '../../../../common/types/common.types';
import { Descendant } from 'slate';
import SlateUtils from '../../../../common/utils/slateUtils';
import CloseIcon from '@mui/icons-material/Close';

type Props = {
  open: boolean;
  note: Note;
  closeDialog: () => void;
};

export const NoteDialog: FC<Props> = ({ note, open, closeDialog }) => {
  const { title, description } = note;
  const [titleValue, setTitleValue] = useState(title);
  const [descriptionValue, setDescriptionValue] = useState(
    SlateUtils.getInitialValueForSlate(description)
  );
  const classes = useNoteDialogStyles();

  const onTitleChange = () => {
    return (e: InputChangeEvent) => setTitleValue(e.target.value);
  };

  const onDescriptionChange = () => {
    return (e: Descendant[]) => setDescriptionValue(e);
  };

  return (
    <Dialog open={open} onClose={closeDialog} fullWidth>
      <DialogTitle className={classes.titleContainer}>
        <IconButton onClick={closeDialog} className={classes.closeIcon}>
          <CloseIcon />
        </IconButton>
      </DialogTitle>

      <DialogContent className={classes.container}>
        <BorderlessInput
          variant="standard"
          value={titleValue}
          onChange={onTitleChange()}
          InputProps={{ style: { fontWeight: 'bold', fontSize: 18 } }}
        />
        <Box className={classes.descriptionContainer}>
          <RichTextField
            value={descriptionValue}
            setValue={onDescriptionChange()}
          />
        </Box>
      </DialogContent>
      <DialogActions>
        <Button color={'primary'} onClick={closeDialog}>
          Save
        </Button>
      </DialogActions>
    </Dialog>
  );
};
