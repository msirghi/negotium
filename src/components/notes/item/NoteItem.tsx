import { Note } from '../../../common/types/notes.types';
import { FC, useState } from 'react';
import {
  Card,
  CardActions,
  CardContent,
  IconButton,
  Tooltip,
  Typography,
} from '@mui/material';
import { useNoteItemStyles } from './styles';
import { NoteDialog } from './dialog/NoteDialog';
import SlateUtils from '../../../common/utils/slateUtils';
import DeleteOutlineIcon from '@mui/icons-material/DeleteOutline';
import DateUtils from '../../../common/utils/dateUtils';
import { NOTE_DATE_FORMAT } from '../../../common/constants/constants';
import useTranslation from 'next-translate/useTranslation';

type Props = {
  note: Note;
  onNoteRemove: (id: Note['id']) => void;
  onNoteUpdate: (note: Note) => void;
};

export const NoteItem: FC<Props> = ({ note, onNoteRemove, onNoteUpdate }) => {
  const { title, description, id, createdDate } = note;
  const classes = useNoteItemStyles();
  const { t } = useTranslation('notes');

  const [dialogOpen, setDialogOpen] = useState(false);

  const openDialog = () => {
    setDialogOpen(true);
  };

  const closeDialog = () => {
    setDialogOpen(false);
  };

  const onDeleteClick = (e: MouseEvent) => {
    e.stopPropagation();
    onNoteRemove(id);
  };

  return (
    <>
      <NoteDialog
        open={dialogOpen}
        note={note}
        closeDialog={closeDialog}
        onNoteUpdate={onNoteUpdate}
      />

      <Card className={classes.container} onClick={openDialog}>
        <CardContent>
          <Typography variant="h5" component="div" className={classes.title}>
            {title}
          </Typography>

          <Typography
            className={classes.description}
            color="text.secondary"
            gutterBottom
          >
            {SlateUtils.serialize(
              SlateUtils.getInitialValueForSlate(description)
            )}
          </Typography>
        </CardContent>
        <CardActions>
          <div className={classes.date}>
            {DateUtils.formatDate(createdDate, NOTE_DATE_FORMAT)}
          </div>
          {/*// @ts-ignore*!/*/}
          <IconButton className={classes.deleteButton} onClick={onDeleteClick} data-testid={'delete-note-btn'}>
            <Tooltip title={t('deleteNote')!}>
              <DeleteOutlineIcon />
            </Tooltip>
          </IconButton>
        </CardActions>
      </Card>
    </>
  );
};
