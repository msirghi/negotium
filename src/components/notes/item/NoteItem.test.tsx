import { NotesMock } from '../../../common/tests/mockData/notes-mock';
import renderer, { act } from 'react-test-renderer';
import { NoteItem } from './NoteItem';
import { mount } from 'enzyme';
import { Card } from '@mui/material';
import { NoteDialog } from './dialog/NoteDialog';

describe('NoteItem', () => {
  const defaultProps = {
    note: NotesMock[0],
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<NoteItem {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should handle dialog open', () => {
    const wrapper = mount(<NoteItem {...defaultProps} />);
    const card = wrapper.find(Card);

    act(() => {
      card.simulate('click');
    });
    wrapper.update();
    expect(wrapper.find(NoteDialog).props().open).toBeTruthy();
  });

  it('should handle dialog close', () => {
    const wrapper = mount(<NoteItem {...defaultProps} />);
    const card = wrapper.find(Card);

    act(() => {
      card.simulate('click');
    });
    wrapper.update();
    const noteDialog = wrapper.find(NoteDialog);
    expect(noteDialog.props().open).toBeTruthy();

    act(() => {
      noteDialog.props().closeDialog();
    });
    wrapper.update();
    expect(wrapper.find(NoteDialog).props().open).toBeFalsy();
  });
});
