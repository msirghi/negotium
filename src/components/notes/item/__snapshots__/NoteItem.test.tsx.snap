// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItem should match snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-container-1 css-bhp9pd-MuiPaper-root-MuiCard-root"
  onClick={[Function]}
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Note 1
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom makeStyles-description-2 css-1ydyjjr-MuiTypography-root"
    >
      new task for 123 
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing makeStyles-cardActions-3 css-1t6e9jv-MuiCardActions-root"
  >
    <div
      className="makeStyles-date-5"
    >
      02 Oct, 2020, 00:00
    </div>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium makeStyles-deleteButton-4 css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DeleteOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </button>
  </div>
</div>
`;
